### PR TITLE
Update frontend components - waiting page, collab page, renew token

### PR DESCRIPTION
**Pull request by yl-ang**

### CODE REVIEW

**Review by jeromewjj — APPROVED** [review]
lgtm

File: frontend/src/components/WaitingPage.js [hunk 1]
```diff
@@ -10,6 +10,7 @@ function WaitingPage() {
   const { userId, socket } = auth;
 
   useEffect(() => {
+    console.log(location.state.difficulty);
```
Comment (deyixtan) [inline frontend/src/components/WaitingPage.js:13]: Should remove

Comment (yl-ang) [reply frontend/src/components/WaitingPage.js:13]: Oops, thanks for highlighting 🔥

**Review by jeromewjj — APPROVED** [review]
Fix merge conflict and lgtm

**Pull request merged**